salt/metalk8s/kubernetes/etcd/installed.sls

```diff
@@ -12,8 +12,12 @@ include:
 {#- Get the list of existing etcd member. #}
 {%- set etcd_members = pillar.metalk8s.etcd.members %}
 
-{#- Compute the initial state according to the existing list of node. #}
-{%- set state = "existing" if etcd_members else "new" %}
+{%- if pillar.is_bootstrap %}
+  {%- set state = 'new' %}
+{%- else %}
+  {%- set state = 'existing' if etcd_members else
+        raise ('etcd member list cannot be empty for existing state')  %}
```

Comment:
No really like the fact to raise in the `set`, keep a `if` `else` block please

Reply:
Just do both at once:

```sls
{%- if pillar.is_bootstrap %}
  {%- set state = 'new' %}
  {%- set etcd_endpoints = {node_name: endpoint} %}
{%- else %}
  {%- set state = 'existing' %}

  {%- set etcd_endpoints = {} %}
  {#- NOTE: Filter out members with empty name as they are not started yet. #}
  {%- for member in etcd_members | selectattr('name') %}
    {#- NOTE: Only take first peer_urls for endpoint. #}
    {%- do etcd_endpoints.update({member['name']: member['peer_urls'][0]}) %}
  {%- else %}
    {{ raise('Br0ken') }}
  {%- endfor %}

  {#- Add ourselves to the endpoints. #}
  {%- do etcd_endpoints.update({node_name: endpoint}) %}
{%- endif %}
```